dcos/http.py

```diff
@@ -1,11 +1,19 @@
+import getpass
+import sys
+
 import requests
 from dcos import util
 from dcos.errors import DCOSException, DCOSHTTPException
+from requests.auth import HTTPBasicAuth
+
+from six.moves.urllib.parse import urlparse
 
 logger = util.get_logger(__name__)
 
 DEFAULT_TIMEOUT = 5
 
+AUTH_CREDS = {}  # (hostname, realm) -> AuthBase()
```

Comment:
Can we verify that we don't have a race condition? I remember that we perform HTTP requests over multiple threads: segment.io and dcos. If this is the case, how do we plan to address those data races? cc @mgummelt.